src/utils/mod.rs

```diff
@@ -1010,3 +1012,154 @@ pub fn create_unrestricted_signing_rsa_public(
+            let selection = &tpml_pcr_selection.pcrSelections[selection_index];
+            let pcr_slots: BitFlags<PcrSlot> =
+                BitFlags::<PcrSlot>::try_from(u32::from_le_bytes(selection.pcrSelect)).unwrap();
+            // Check for variations in sizeofSelect.
+            // Something that currently is not supported.
+            if selection.sizeofSelect != size_of_select.unwrap_or(selection.sizeofSelect) {
+                return Err(Error::local_error(WrapperErrorKind::InconsistentParams));
+            }
+            size_of_select = Some(selection.sizeofSelect);
+            let _ = ret.items.insert(
+                HashingAlgorithm::try_from(selection.hash).unwrap(),
+                pcr_slots,
+            );
+        }
+        ret.size_of_select = size_of_select.unwrap();
+        Ok(ret)
+    }
+}
+
+/// A builder for the PcrSelection struct.
+#[derive(Debug, Default)]
+pub struct PcrSelectionsBuilder {
+    size_of_select: Option<u8>,
+    items: HashMap<HashingAlgorithm, BitFlags<PcrSlot>>,
+}
+
+impl PcrSelectionsBuilder {
+    pub fn new() -> Self {
+        PcrSelectionsBuilder {
+            size_of_select: None,
+            items: Default::default(),
+        }
+    }
+
+    /// Set the size of the pcr selection(sizeofSelect)
+    ///
+    /// # Arguments
+    /// size_of_select -- The size that will be used for all selections(sizeofSelect).
+    pub fn with_size_of_select(mut self, size_of_select: u8) -> Self {
+        self.size_of_select = Some(size_of_select);
+        self
+    }
+
+    /// Adds a selection associated with a specific HashingAlgorithm.
+    ///
+    /// This function will not overwrite the values already associated
+    /// with a specific HashingAlgorithm only update.
+    ///
+    /// # Arguments
+    /// hash_algorithm -- The HashingAlgorithm associated with the pcr selection
+    /// pcr_slots -- The PCR slots in the selection.
+    pub fn with_selection(
+        mut self,
+        hash_algorithm: HashingAlgorithm,
+        pcr_slots: &[PcrSlot],
+    ) -> Self {
+        let selected_pcr_slots: BitFlags<PcrSlot> = pcr_slots.iter().cloned().collect();
+        match self.items.get_mut(&hash_algorithm) {
+            Some(previously_selected_pcr_slots) => {
+                *previously_selected_pcr_slots |= selected_pcr_slots;
+            }
+            None => {
+                let _ = self.items.insert(hash_algorithm, selected_pcr_slots);
+            }
+        }
+        self
+    }
+
+    /// Builds a PcrSelections with the values that have been
+    /// provided.
+    ///
+    /// If no size of select have been provided then it will
+    /// be defaulted to 3. This may not be the correct size for
+    /// the current platform. The correct values can be obtained
+    /// by quering the tpm for its capabilities.
+    pub fn build(self) -> PcrSelections {
+        let select_size = match self.size_of_select {
+            Some(value) => value,
+            None => 3,
```

Comment:
Do you think that's fine to put 3 there? We could also return an error if this value not set.

Reply:
3 covers the slots specified by the enum. Future proof value according to the specification is 4. But right now I do not think that there is any one using that.

Returning an error would be nice if people using this knows what the value should be. Would it be enough to just put a recommended value in the documentation?

Reply:
We could also do something like this 

```rust
pub enum SelectSize {
One_Byte = 1,
Two_Bytes = 2,
Three_Bytes = 3,
Four_Bytes = 4,
}

impl Default for SelectSize {
fn default() -> SelecSize {
// Three bytes is a commonly used value
// That work on most platforms.
SelectSize::Three_Bytes
}
}
```

Reply:
I am fine with just using 3 in that case! I was just worried where it came from. It might be nice to put it in a `const` value with a comment saying what you wrote: "Three bytes is a commonly used value that works on most plaftorms."

Reply:
I kind of like the enum approach because it makes it harder to send in a value that is completely wrong. So i think it will be safer that way.

Reply:
Are those four values the only ones allowed?

Reply:
Yes. They dictate the size of a variable bit mask. The buffer for the bit mask is [4: u8]. With the sizeofSelect parameter to tell how many bytes to use. So no other values makes any sense.

Reply:
Great! Then the enum with the `Default` seems like a good option to me!